posthog/hogql/printer.py

```diff
@@ -0,0 +1,213 @@
+            response = "true"
+        elif isinstance(node.value, bool) and node.value is False:
+            response = "false"
+        elif isinstance(node.value, int) or isinstance(node.value, float):
+            # :WATCH_OUT: isinstance(True, int) is True (!), so check for numbers lower down the chain
+            response = str(node.value)
+        elif isinstance(node.value, str) or isinstance(node.value, list):
+            context.values[key] = node.value
+            response = f"%({key})s"
+        elif node.value is None:
+            response = "null"
+        else:
+            raise ValueError(
+                f"Unknown AST Constant node type '{type(node.value).__name__}' for value '{str(node.value)}'"
+            )
+    elif isinstance(node, ast.Field):
+        if dialect == "hogql":
+            # When printing HogQL, we print the properties out as a chain instead of converting them to Clickhouse SQL
+            response = ".".join([print_hogql_identifier(identifier) for identifier in node.chain])
+        elif node.chain == ["*"]:
+            query = f"tuple({','.join(SELECT_STAR_FROM_EVENTS_FIELDS)})"
+            response = print_ast(parse_expr(query), stack, context, dialect)
+        elif node.chain == ["person"]:
+            query = "tuple(distinct_id, person.id, person.created_at, person.properties.name, person.properties.email)"
+            response = print_ast(parse_expr(query), stack, context, dialect)
+        else:
+            field_access = parse_field_access(node.chain, context)
+            context.field_access_logs.append(field_access)
+            response = field_access.sql
+    elif isinstance(node, ast.Call):
+        if node.name in HOGQL_AGGREGATIONS:
+            context.found_aggregation = True
+            required_arg_count = HOGQL_AGGREGATIONS[node.name]
+
+            if required_arg_count != len(node.args):
+                raise ValueError(
+                    f"Aggregation '{node.name}' requires {required_arg_count} argument{'s' if required_arg_count != 1 else ''}, found {len(node.args)}"
+                )
+
+            # check that we're not running inside another aggregate
+            for stack_node in stack:
+                if stack_node != node and isinstance(stack_node, ast.Call) and stack_node.name in HOGQL_AGGREGATIONS:
+                    raise ValueError(
+                        f"Aggregation '{node.name}' cannot be nested inside another aggregation '{stack_node.name}'."
+                    )
+
+            translated_args = ", ".join([print_ast(arg, stack, context, dialect) for arg in node.args])
+            if dialect == "hogql":
+                response = f"{node.name}({translated_args})"
+            elif node.name == "count":
+                response = "count(*)"
+            elif node.name == "countDistinct":
+                response = f"count(distinct {translated_args})"
+            elif node.name == "countDistinctIf":
```

Comment:
Are we keeping a list of all the little dialect diversions anywhere?

Reply:
Nope. This needs to be redone. It should be the standard dialect here.